sql/migrate/migrate.go

```diff
@@ -574,6 +609,39 @@ func (r *Revision) setSQLErr(err error, stmt string) error {
 	return err
 }
 
+const (
+	// KindExecuting is sent once when execution of the migration files has been started.
+	KindExecuting LogKind = iota + 1
+	// KindFile is sent if a new migration file is executed.
+	KindFile
+	// KindStmt is sent if a new SQL statement is executed.
+	KindStmt
+)
+
+type (
+	// LogKind denotes the kind of a LogEntry.
+	LogKind uint
+
+	// An LogEntry represents one entry written to a configured Logger while migration execution is running.
+	LogEntry struct {
+		// Kind of this entry.
+		Kind LogKind
+		// Pending files to be executed.
+		Pending []string
+		// Version executed.
+		Version string
+		// Desc of migration executed.
+		Desc string
+		// SQL statement executed.
+		SQL string
+	}
```

Comment:
I wonder maybe define "LogEntry" as an interface and let the logger print it with type assertion. A good example is the `ast.Node` in the different `ast` packages (go, templates, graphql, etc).